code/modules/mob/living/basic/blob_minions/blob_zombie.dm

```diff
@@ -0,0 +1,97 @@
+	name = "blob zombie"
+	desc = "A shambling corpse animated by the blob."
+	icon_state = "zombie"
+	icon_living = "zombie"
+	health_doll_icon = "blobpod"
+	mob_biotypes = MOB_ORGANIC | MOB_HUMANOID
+	health = 70
+	maxHealth = 70
+	verb_say = "gurgles"
+	verb_ask = "demands"
+	verb_exclaim = "roars"
+	verb_yell = "bellows"
+	melee_damage_lower = 10
+	melee_damage_upper = 15
+	obj_damage = 20
+	attack_verb_continuous = "punches"
+	attack_verb_simple = "punch"
+	attack_sound = 'sound/weapons/genhit1.ogg'
+	death_message = "collapses to the ground!"
+	gold_core_spawnable = NO_SPAWN
+	basic_mob_flags = DEL_ON_DEATH
+	ai_controller = /datum/ai_controller/basic_controller/blob_zombie
+	/// The dead body we have inside
+	var/mob/living/carbon/human/corpse
+
+/mob/living/basic/blob_minion/zombie/Initialize(mapload)
+	. = ..()
+	AddElement(/datum/element/swabable, CELL_LINE_TABLE_BLOBSPORE, CELL_VIRUS_TABLE_GENERIC_MOB, 1, 5)
+
+/mob/living/basic/blob_minion/zombie/death(gibbed)
+	corpse?.forceMove(loc)
+	death_burst()
+	return ..()
+
+/mob/living/basic/blob_minion/zombie/Exited(atom/movable/gone, direction)
+	. = ..()
+	if (gone != corpse)
+		return
+	corpse = null
+	death()
+
+/mob/living/basic/blob_minion/zombie/Destroy()
+	QDEL_NULL(corpse)
+	return ..()
+
+/mob/living/basic/blob_minion/zombie/on_factory_destroyed()
+	. = ..()
+	death()
+
+/mob/living/basic/blob_minion/zombie/update_overlays()
+	. = ..()
+	copy_overlays(corpse, TRUE)
+	var/mutable_appearance/blob_head_overlay = mutable_appearance('icons/mob/nonhuman-player/blob.dmi', "blob_head")
+	blob_head_overlay.color = LAZYACCESS(atom_colours, FIXED_COLOUR_PRIORITY) || COLOR_WHITE
+	color = initial(color) // reversing what our component did lol, but we needed the value for the overlay
+	. += blob_head_overlay
+
+/// Create an explosion of spores on death
+/mob/living/basic/blob_minion/zombie/proc/death_burst()
+	do_chem_smoke(range = 0, holder = src, location = get_turf(src), reagent_type = /datum/reagent/toxin/spore)
+
+/// Store a body so that we can drop it on death
+/mob/living/basic/blob_minion/zombie/proc/consume_corpse(mob/living/carbon/human/new_corpse)
+	if(new_corpse.wear_suit)
+		maxHealth += new_corpse.get_armor_rating(MELEE)
+		health = maxHealth
+	new_corpse.set_facial_hairstyle("Shaved", update = FALSE)
+	new_corpse.set_hairstyle("Bald", update = TRUE)
+	new_corpse.forceMove(src)
+	corpse = new_corpse
+	update_appearance(UPDATE_ICON)
+	RegisterSignal(corpse, COMSIG_LIVING_REVIVE, PROC_REF(on_corpse_revived))
+
+/// Dynamic changeling reentry
+/mob/living/basic/blob_minion/zombie/proc/on_corpse_revived()
+	SIGNAL_HANDLER
+	visible_message(span_boldwarning("[src] bursts from the inside!"))
+	death()
+
+/// Blob-created zombies will ping for player control when they make a zombie
+/mob/living/basic/blob_minion/zombie/controlled
+
+/mob/living/basic/blob_minion/zombie/controlled/consume_corpse(mob/living/carbon/human/new_corpse)
+	. = ..()
+	if (key)
```

Comment:
change this to a `!isnull(client)` check? i think a blob zombie freeing up if someone disconnects would be considered as useful to the blob. it's okay if you're not willing to change it though